### PR TITLE
bugfix src/widget_window.c for running X under cage

### LINKED ISSUE #106: gtkdialog throws warnings under X under cage
See https://github.com/puppylinux-woof-CE/woof-CE/issues/2670

An attempt was made to fix in #105 

I'm going to try `getenv(WAYLAND_DISPLAY)` _but_ it still might throw the warnings under `cage` running X as a child.

We'll see.

### PR DESCRIPTION
**Pull request by 01micko**
closes #106

### CODE REVIEW

**Comment by 01micko** [comment]
Should close https://github.com/puppylinux-woof-CE/woof-CE/issues/2670

If/when this PR is merged I'll release 0.8.4h

File: src/widget_window.c [hunk 1]
```diff
@@ -98,6 +98,12 @@ GtkWidget *widget_window_create(
 	widget = gtk_window_new(GTK_WINDOW_TOPLEVEL);  
 
 #if HAVE_GTK_LAYER_SHELL
+	if (!getenv("WAYLAND_DISPLAY"))
```
Comment (dimkr) [inline src/widget_window.c:101]: Some GTK+ applications run with `GDK_BACKENDS=x11` (mtpaint forces this on itself) ... this is not the right way to check if the Wayland backend is **in use**.

Comment (dimkr) [reply src/widget_window.c:101]: btw, WAYLAND_DISPLAY is set under Cage

Comment (01micko) [reply src/widget_window.c:101]: > btw, WAYLAND_DISPLAY is set under Cage
I know, that's why the second test.

Comment (01micko) [reply src/widget_window.c:101]: What's the best way to test the value of `GDK_BACKEND` in the context of the rest of the gtkdialog code? Plain C or glib? I see a mixture here and there.

Comment (dimkr) [reply src/widget_window.c:101]: Try https://github.com/puppylinux-woof-CE/gtkdialog/issues/106#issuecomment-991842337

**Pull request closed without merging**